src/daal4py.cpp

```diff
@@ -602,28 +551,20 @@ daal::data_management::NumericTablePtr make_nt(PyObject * obj)
                     DAAL4PY_CHECK_MALLOC(c_roffs_one_based);
                     for (size_t i = 0; i < n; ++i) c_roffs_one_based[i] = c_roffs[i] + 1;
                     size_t c_nc = static_cast<size_t>(PyInt_AsSsize_t(nc));
-                    if (PyErr_Occurred())
-                    {
-                        PyErr_Print();
-                        throw std::runtime_error("Python Error");
-                    }
+                    py_err_check();
                     size_t c_nr = static_cast<size_t>(PyInt_AsSsize_t(nr));
-                    if (PyErr_Occurred())
-                    {
-                        PyErr_Print();
-                        throw std::runtime_error("Python Error");
-                    }
+                    py_err_check();
 #define MKCSR_(_T) ret = daal::data_management::CSRNumericTable::create(daal::services::SharedPtr<_T>(reinterpret_cast<_T *>(array_data(np_vals)), NumpyDeleter(reinterpret_cast<PyArrayObject *>(np_vals))), daal::services::SharedPtr<size_t>(c_indcs_one_based, daal::services::ServiceDeleter()), daal::services::SharedPtr<size_t>(c_roffs_one_based, daal::services::ServiceDeleter()), c_nc, c_nr)
-                    SET_NPY_FEATURE(array_type(np_vals), MKCSR_, throw std::invalid_argument("Found unsupported data type in csr_matrix"));
+                    SET_NPY_FEATURE(array_type(np_vals), MKCSR_, throw std::invalid_argument("Found unsupported data type in CSR object"));
```

Comment:
I think "CSR object" can be a misleading term. Maybe it will be better to explicitly specify "csr_matrix or csr_array".